shell/platform/embedder/tests/embedder_frozen_unittests.cc

```diff
@@ -0,0 +1,97 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/embedder/embedder.h"
+#include "flutter/shell/platform/embedder/tests/embedder_frozen.h"
+#include "flutter/testing/testing.h"
+
+namespace flutter {
+namespace testing {
+
+// New members must not be added to `FlutterSemanticsNode`
+// as it would break the ABI of `FlutterSemanticsUpdate`.
+// See: https://github.com/flutter/flutter/issues/121176
+TEST(EmbedderFrozen, FlutterSemanticsNodeIsFrozen) {
+  ASSERT_EQ(sizeof(FlutterSemanticsNode), sizeof(FrozenFlutterSemanticsNode));
+
+  ASSERT_EQ(offsetof(FlutterSemanticsNode, struct_size),
+            offsetof(FrozenFlutterSemanticsNode, struct_size));
```

Comment:
@chinmaygarde may have thoughts on whether there's some way to use template metaprogramming to iterate over the fields of a struct, but this one is definitely beyond my ken.

Reply:
Yep something like `ASSERT_EQ_OFFSET(EmbedderStruct, FrozenEmbedderStruct, field)` would be a nice cleanup.

Reply:
Excellent idea, updated!